testing/testcontainers/java/src/test/java/io/crate/example/testing/TestSharedSingletonEnvironmentVersion.java

```diff
@@ -0,0 +1,65 @@
+package io.crate.example.testing;
+
+import org.junit.Test;
+import org.testcontainers.cratedb.CrateDBContainer;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+import static io.crate.example.testing.utils.TestingHelpers.assertResults;
+
+
+/**
+ * Testcontainer instance honoring the `CRATEDB_VERSION` environment variable,
+ * suitable for running a test matrix on different versions of CrateDB.
+ * <p>
+ * Possible values are:
+ * <ul>
+ *     <li>Version numbers: 5, 5.2, 5.2.3, etc.</li>
+ *     <li>Latest release: latest</li>
+ *     <li>Nightly release: nightly</li>
+ * </ul>
+ * </p>
+ */
+abstract class AbstractContainerSingletonEnvironmentVersion {
+
+    static final CrateDBContainer cratedb;
+
+    static {
+
+        String cratedb_version = System.getenv("CRATEDB_VERSION");
+        String fullImageName;
+        if (cratedb_version == null) {
+            fullImageName = "crate:latest";
+        } else {
+            if (cratedb_version.equals("nightly")) {
+                fullImageName = "crate/crate:nightly";
+            } else {
+                fullImageName = String.format("crate:%s", cratedb_version);
+            }
+        }
+
+        // Run designated CrateDB version.
+        DockerImageName image = DockerImageName.parse(fullImageName).asCompatibleSubstituteFor("crate");
+        cratedb = new CrateDBContainer(image);
+        cratedb.start();
+    }
+}
```

Comment:
Thanks for adding a `TestingHelpers.java` already.

I think the gist of `AbstractContainerSingletonEnvironmentVersion` may be refactored there in form of a helper method returning the proper `DockerImageName` instance, in order to reuse it on other spots.

Otherwise, when users are only looking at specific test case variants, they may miss that we offer a convenient helper method for this detail. I will outline the other spots with corresponding comments.

Maybe `TestingHelpers.getCrateDBImageName()` would be an appropriate name for that helper method? Or put it into `CrateDBContainer.getImageName()` instead, if that would even be possible?

Reply:
Maybe it is sensible to split it, in order to be reusable in different contexts. If it does not fit into `CrateDBContainer` or `TestingHelpers`, maybe it can go into a new `ImageHelpers.java`?

- `ImageHelpers.nameFromLabel(label)`
- `ImageHelpers.nameFromEnvironment()`

```java
DockerImageName nameFromLabel(label) {
  String fullImageName;
  if (label == null) {
      fullImageName = "crate:latest";
  } else {
      if (label.equals("nightly")) {
          fullImageName = "crate/crate:nightly";
      } else {
          fullImageName = String.format("crate:%s", label);
      }
  }
  return DockerImageName.parse(fullImageName).asCompatibleSubstituteFor("crate");
}
```
```java
DockerImageName nameFromEnvironment() {
  String label = System.getenv("CRATEDB_VERSION");
  return nameFromLabel(label);
}
```

Reply:
Yep, you can do it, use `public static` for those methods, so you can call them without instantiating the class. I'd say just add them to the TestingHelpers class, no need to create a new one.

Reply:
Pushed this change.